cfme/tests/cloud_infra_common/test_genealogy.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme.cloud.instance import EC2Instance, OpenStackInstance, edit_form
+from cfme.infrastructure.virtual_machines import Vm
+from utils import testgen
+from utils.providers import is_cloud_provider
+
+pytestmark = [
+    pytest.mark.usefixtures('uses_infra_providers', 'uses_cloud_providers', 'provider_type')
+]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.all_providers(
+        metafunc, 'small_template', template_location=["small_template"])
+
+    new_idlist = []
+    new_argvalues = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+
+        if metafunc.function in {test_vm_genealogy_detected} \
+                and args["provider_type"] in {"scvmm", "rhevm"}:
+            continue
+
+        new_idlist.append(idlist[i])
+        new_argvalues.append(argvalues[i])
+
+    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
+
+
+@pytest.fixture(scope="function")
+def vm_name():
+    return 'test_genealogy_{}'.format(fauxfactory.gen_alpha(length=8).lower())
+
+
+@pytest.fixture(scope="function")
```

Comment:
I feel that maybe we will be doing more of this? Should we hare a reverser helper somewhere? Or maybe a property of the provider_crud that gives us instance class objects?